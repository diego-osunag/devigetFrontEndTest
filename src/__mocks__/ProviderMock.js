import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import stateMock from './stateMock';
import reducer from '../reducers';

const store = createStore(reducer, stateMock);

const ProviderMock = (props) => (
  <Provider store={store}>
    {props.children}
  </Provider>
);

export default ProviderMock;
