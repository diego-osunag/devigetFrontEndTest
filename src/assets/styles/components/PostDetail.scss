.postDetail {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  height: 100vh;
  transition: 350ms;
  overflow-y: scroll;
}

.postDetail__author {
  color: #151515;
  font-weight: normal;
  text-align: center;
  padding: 50px 0 20px 0;
  margin: 0;
  transition: 350ms;
}

.postDetail__img {
  cursor: pointer;
}

.postDetail__title {
  text-align: center;
  font-size: 1.25em;
  margin: 20px 0;
  width: 90%;
  transition: 350ms;
}

.postDetail__description {
  text-align: left;
  margin: 0 10px 40px 10px;
  width: 90%;
  transition: 350ms;
  word-break: break-all;
}

.postDetail__fullImg {
  position: fixed;
  height: 70vh;
  width: auto;
  z-index: 4; 
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
   visibility: hidden;
   opacity: 0; 
   transition: visibility 300ms linear, opacity 300ms ease-in-out; 
  cursor: pointer;
  box-shadow: 0 14px 28px rgba(0,0,0,0.25), 0 10px 10px rgba(0,0,0,0.22);
}

.postDetail__fullImg.active {
   visibility: visible; 
   opacity: 1; 
}