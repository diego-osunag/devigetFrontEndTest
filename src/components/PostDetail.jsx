import React, { useState } from 'react';
import { connect } from 'react-redux';
import '../assets/styles/components/PostDetail.scss';

const PostDetail = ({ selectedPost }) => {
  const [isActive, setActive] = useState();

  const validateThumbUrl = () => {
    switch (selectedPost.thumbnail) {
      case 'self':
        return '';
      case 'default':
        return '';
      default:
        return selectedPost.thumbnail;
    }
  };

  const validateFullImgUrl = () => (
    selectedPost.preview != null ? 
      selectedPost.preview.images[0].source.url.replace(/&amp;/g, '&')
      : ''
  );

  const handleToggle = () => {
    setActive(!isActive);
  };

  return (
    <div className='postDetail'>
      <h2 className='postDetail__author'>{ selectedPost.author }</h2>
      <img className='postDetail__img' src={validateThumbUrl()} onClick={handleToggle} alt='' />
      <h1 className='postDetail__title'>{ selectedPost.title }</h1>
      <p className='postDetail__description'>{ selectedPost.selftext }</p>
      <img
        className={isActive ? 'postDetail__fullImg active' : 'postDetail__fullImg'}
        src={validateFullImgUrl()}
        onClick={handleToggle}
        alt=''
      />
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    selectedPost: state.selectedPost,
  };
};

export default connect(mapStateToProps, null)(PostDetail);
