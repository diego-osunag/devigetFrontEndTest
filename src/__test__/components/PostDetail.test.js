import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';
import ProviderMock from '../../__mocks__/ProviderMock';
import PostDetail from '../../components/PostDetail';

describe('<PostDetail />', () => {
  const post = mount(
    <ProviderMock>
      <PostDetail />
    </ProviderMock>,
  );

  test('PostDetail component rendering', () => {
    expect(post.length).toEqual(1);
  });

  test('Author\'s name rendering', () => {
    expect(post.find('h2.postDetail__author').text()).toEqual('0-07i');
  });

  test('Thumbnail image rendering', () => {
    expect(post.find('img.postDetail__img').length).toEqual(1);
  });

  test('Post title rendering', () => {
    expect(post.find('h1.postDetail__title').text()).toEqual('How do I find external services in the code?');
  });
});

describe('<PostDetail /> Snapshot', () => {
  test('PostDetail UI rendering correctly', () => {
    const post = create(
      <ProviderMock>
        <PostDetail />
      </ProviderMock>,
    );
    expect(post.toJSON()).toMatchSnapshot();
  });
});
