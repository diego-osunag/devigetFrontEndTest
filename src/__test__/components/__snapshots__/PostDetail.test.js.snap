// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostDetail /> Snapshot PostDetail UI rendering correctly 1`] = `
<div
  className="postDetail"
>
  <h2
    className="postDetail__author"
  >
    0-07i
  </h2>
  <img
    alt=""
    className="postDetail__img"
    onClick={[Function]}
    src="https://a.thumbs.redditmedia.com/08MUc3eyJ4zs2f29UIAG53tGKMU6KwOFTb1An2aGUJ4.jpg"
  />
  <h1
    className="postDetail__title"
  >
    How do I find external services in the code?
  </h1>
  <p
    className="postDetail__description"
  >
    This might be a stupid question...    I am trying to narrow down any external services used by the UI, how do I find it where do I look for them?  Thank you sm!
  </p>
  <img
    alt=""
    className="postDetail__fullImg"
    onClick={[Function]}
    src="https://external-preview.redd.it/ftBi-vGkAc8oHoWPuMqxyrHV7e7TSVRxkAgr3FZ2NbA.jpg?auto=webp&s=3791ed9043209bf57f16f624d8a12471e3e1f748"
  />
</div>
`;
